minmea.h

```diff
@@ -58,13 +58,17 @@ struct minmea_time {
     int microseconds;
 };
 
-struct minmea_type {
+union minmea_type {
+  char buf[6];
```

Comment:
Might be worth a comment here, since absent the discussion in the PR, it might not be obvious why there's a 6 byte buf here. Maybe something like `//provide backwards compatibility to users expecting a null-terminated string instead of a struct`.